readhomer_atlas/library/models.py

```diff
@@ -0,0 +1,76 @@
+from django.db import models
+from django.contrib.postgres.fields import JSONField
+
+
+class Version(models.Model):
+    """
+    urn:cts:greekLit:tlg0012.tlg001.perseus-grc2
+    """
+
+    urn = models.CharField(max_length=255)
+    name = models.CharField(blank=True, null=True, max_length=255)
+    metadata = JSONField(encoder="", default=dict, blank=True)
+    """
+    {
+        "work_urn": "urn:cts:greekLit:tlg0012.tlg001",
+        "work_title": "Iliad",
+        "type": "edition"
+    }
+    """
+
+    class Meta:
+        ordering = ["urn"]
+
+    def __str__(self):
+        return self.name
+
+
+class Book(models.Model):
```

Comment:
Just thinking aloud here, but in the absence of any book-level properties other than an identifier, is there any reason to even have this model as opposed to lines just having a book field which is indexed?

(Same applies to lines if we have a token model)

Reply:
I can see three approaches:

1) the above approach.  We use the FKs to provide the relationship of lines to books.  Following this approach, I’d likely have a FK to a 'Card` model too.

I chose this as a starting point because it lends itself really well to the nested GraphQL queries.

2) The indexed field approach. We’d probably need to think about how to structure the GraphQL queries and the graphene backend so you could query to see all of the books.

3) 1 + 2. Denorm using a `book_number` field, but keep a `Book` FK.

You might also be able to write the GraphQL query to implement 2 using the modeling of 1.